### PR TITLE
Use simpler check for authenticated user

### PR DESCRIPTION
**Pull request by labkey-matthewb**
#### Rationale
AuthenticatedRequest.isGuestSession() returns false for robots causing us to track sessions that won't ever fire SessionListener.sessionDestroyed()

#### Related Pull Requests
- 

#### Changes
-

### CODE REVIEW

**Comment by labkey-matthewb** [comment]
Not right yet.  It seems we do want to track "real" guests?

**Comment by labkey-matthewb** [comment]
I think AuthenticatedRequest.isGuestSession() isn't very meaningful here.  I think these changes are correct if the intention is that getActiveUserSessionCount() returns the count authenticated users sessions.  If that's not the case, then I'm confused about what the previous code was trying to do.

**Review by labkey-jeckels — APPROVED** [review]

**Review by labkey-jeckels — CHANGES_REQUESTED** [review]
The new assert on line 509 is failing.

File: api/src/org/labkey/api/security/UserManager.java [hunk 1]
```diff
@@ -506,14 +500,24 @@ public static int getUniqueUsersCount(Date since)
...
 
-    private static final Map<String, HttpSession> _activeSessions = Collections.synchronizedMap(new HashMap<>());
+    private static final Set<String> _activeSessions = Collections.synchronizedSet(new HashSet<>());
+
+    public static void ensureSessionTracked(HttpSession s)
+    {
+        if (s != null)
+        {
+            Integer userId = (Integer)s.getAttribute(USER_ID_KEY);
+            if (null != userId && 0 != userId)
```
Comment (labkey-jeckels) [inline api/src/org/labkey/api/security/UserManager.java]:
```suggestion
            if (null != userId && getGuestUser().getUserId() != userId)
```

**Review by labkey-jeckels — APPROVED** [review]

**Pull request merged**